apollo-router/src/configuration/schema.rs

```diff
@@ -32,48 +26,18 @@ use crate::configuration::upgrade::upgrade_configuration;
 
 const NUMBER_OF_PREVIOUS_LINES_TO_DISPLAY: usize = 5;
 
-/// This needs to exist because Schemars incorrectly generates references with spaces in them.
-/// We just rename them.
-#[derive(Debug, Clone)]
-struct RefRenameVisitor;
-
-impl Visitor for RefRenameVisitor {
-    fn visit_root_schema(&mut self, root: &mut RootSchema) {
-        visit_root_schema(self, root);
-        root.definitions = mem::take(&mut root.definitions)
-            .into_iter()
-            .map(|(k, v)| (k.replace(' ', "_"), v))
-            .collect();
-    }
-    fn visit_schema_object(&mut self, schema: &mut SchemaObject) {
-        if let Some(reference) = &mut schema.reference {
-            schema.metadata = Some(Box::new(Metadata {
-                description: Some(reference.clone()),
-                ..Default::default()
-            }));
-            *reference = reference.replace(' ', "_");
-        }
-
-        visit_schema_object(self, schema);
-    }
-}
-
 /// Generate a JSON schema for the configuration.
-pub(crate) fn generate_config_schema() -> RootSchema {
+pub(crate) fn generate_config_schema() -> Schema {
     let settings = SchemaSettings::draft07().with(|s| {
-        s.option_nullable = true;
-        s.option_add_null_type = false;
```

Comment:
This difference is worth calling out. In the new schemars i believe this option is gone. This means that instead of `nullable: true`, you get `type: [object, null]` or `anyOf: [{ type: object }, { type: null }]`. In my manual testing I did not find this a regression for the editing experience, but it does make error messages a bit worse.

I _think_ that generating a newer version than draft-07 would enable using `nullable: true` again, as it isn't strictly part of that?

Reply:
Ah, `nullable: true` is just not a JSON Schema thing, but an openapi thing. This seems right then honestly.